org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -624,6 +626,22 @@ public void doGenerate(Resource resource, LFGeneratorContext context) {
                 Exceptions.sneakyThrow(e);
             }
 
+            // Dump the additional compile definitions to a file to keep the generated project
+            //  self contained. In this way, third-party build tools like PlatformIO, west, arduino-cli can
+            //  take over and do the rest of compilation.
+
+            try {
+                String compileDefs = targetConfig.compileDefinitions.keySet().stream()
+                    .map(key -> key + "=" + targetConfig.compileDefinitions.get(key))
+                    .collect(Collectors.joining("\n"));
+                FileUtil.writeToFile(
+                    compileDefs,
+                    Path.of(fileConfig.getSrcGenPath() + File.separator + "CompileDefinitions.txt")
+                );
+            } catch (IOException e) {
+                Exceptions.sneakyThrow(e);
```

Comment:
What is this `sneakyThrow` for? I've seen these starting to appear since the port from Xtend to Java, which makes me believe this was used in the Xtend-generated Java code.

Reply:
I dont know. I copied that pattern from a few lines above.

Reply:
https://www.baeldung.com/java-sneaky-throws